application/models/Token.php

```diff
@@ -231,13 +234,34 @@ public function findByToken($token)
         ));
     }
 
+    /**
+     * Get survey token length from survey. 
+     * Use default if not possible.
+     */
+    public function getSurveyTokenLength()
+    {
+        // Use default token length
+        $iTokenLength = self::DEFAULT_LENGTH;        
+
+        // Use survey token length, if defined
+        if (isset($this->survey) && !empty($this->survey->oOptions) && !empty($this->survey->oOptions->tokenlength) && is_numeric($this->survey->oOptions->tokenlength)) {
+            $iTokenLength = $this->survey->oOptions->tokenlength;
+        }
+
+        return $iTokenLength;
+    }
+
     /**
      * Generates a token for this object.
      * @throws CHttpException
      */
     public function generateToken($tokenlength = NULL)
     {
-        $iTokenLength = $tokenlength ? $tokenlength : $this->survey->tokenlength;
+        if ($tokenlength) {
```

Comment:
Why not `generateToken($iTokenLength = NULL)`

And 
```
if(empty($iTokenLength)) {
    $iTokenLength = $this->getSurveyTokenLength();
}
```

I don't understand why use 2 different parameter name here ....

Reply:
You are right.